### PR TITLE
Made deep RL model

### PR DESCRIPTION
**Pull request by sane-RK9**
I have created an deep RL model #12  using tenserflow and sklearn for data processing. 
I have will start creating the data files and edit the program to handle those Files for data.
Plz confirm if the model if is good to go and doesn't have potential issues.

### CODE REVIEW

**Comment by RAGHAVS1304** [comment]
Hi @sane-RK9, the approach looks quite good and effectively handles real-time factors as well. Have you tried running it on a real dataset to evaluate its performance? It would be helpful to see the metric values (in a notebook). Also, based on that, you can determine whether to move forward with DQN.

**Comment by sane-RK9** [comment]
actually there's issue in it it uses 3 loops in step function which causes the training to slowdown i was thinking of using multiprocessing but still haven't learned that.
Coming to dataset training of thinking of using synthetic(using faker and sdv) data  for now.
for now

**Comment by RAGHAVS1304** [comment]
If it's feasible to find a real dataset, try it; otherwise synthetic one will also work for now.

**Comment by sane-RK9** [comment]
My apologies for keeping it hanging i have added code into notebook(its buggy for now apologies) with pytorch and performance metrices class and in the data folder is the data i found therefore. I'll add try adding a more version cleaned version with synthetic data by tomorrow. I try to keep you guys updated.

**Comment by sane-RK9** [comment]
Here the model finally complete now the coming to data I tried with synthetic it has some issues so didn't go with it. I have changed the model to from dqn to td-3 since I saw the model was not gonna work for continuous change in pricing. Secondly in the data the aircraft type column is not present which is useful to find how much plan in flight costs for per value but since that info is not available in the kraggle data . I hope model and notebook is acceptable . My apologies for underestimating the challenges of synthetic data .

**Comment by RAGHAVS1304** [comment]
@sane-RK9 I didn't find the outputs of your model. It would be better if you can share some screenshots showing the predictions made by your model (if it is completed). 
Also, if it is incomplete or not working, then just do one thing, explain your entire approach (in detail) that you thought of, we will accept that document, as a part of future scope.

**Comment by RAGHAVS1304** [comment]
@sane-RK9 do the above before 7th April, so that the document or write-up could be merged.

**Comment by sane-RK9** [comment]
[documentation.docx](https://github.com/user-attachments/files/19622718/documentation.docx)
i am still lacking in documentation but here's the it complete explaning the complete notebook and alorigthm i hope it helps highilight issues due to data i was incapable to find in good format from kraggle

**Comment by krittika2004** [comment]
Hi @sane-RK9 
Can u please raise the PR again with a different branch cuz u have raised the PR using main branch and there are some merge conflicts. Plz do it quickly so that I can merge it.

**Comment by krittika2004** [comment]
@sane-RK9 
the base branch is the branch you want to merge your changes into.
Pl. change base branch back to main.. And raise your PR with a different branch
